### ISSUE TITLE
cannot bind properties

### ISSUE THREAD

**Issue by spencergibb** [spring-cloud/spring-cloud-function#276]
https://jenkins.spring.io/view/Spring%20Cloud/view/CI/job/spring-cloud-function-master-ci/2327/testReport/

`org.springframework.cloud.function.context.HybridFunctionalRegistrationTests.testNoDoubleRegistrationInHybridMode`

```
java.lang.IllegalStateException: Cannot bind properties
	at org.springframework.cloud.function.context.HybridFunctionalRegistrationTests.testNoDoubleRegistrationInHybridMode(HybridFunctionalRegistrationTests.java:43)
Caused by: org.springframework.beans.factory.NoSuchBeanDefinitionException: No bean named 'org.springframework.boot.context.internalConfigurationPropertiesBinder' available
	at org.springframework.cloud.function.context.HybridFunctionalRegistrationTests.testNoDoubleRegistrationInHybridMode(HybridFunctionalRegistrationTests.java:43)
```
Logs:
```
  .   ____          _            __ _ _
 /\\ / ___'_ __ _ _(_)_ __  __ _ \ \ \ \
( ( )\___ | '_ | '_| | '_ \/ _` | \ \ \ \
 \\/  ___)| |_)| | | | | || (_| |  ) ) ) )
  '  |____| .__|_| |_|_| |_\__, | / / / /
 =========|_|==============|___/=/_/_/_/
 :: Spring Boot ::  (v2.2.0.BUILD-SNAPSHOT)

2019-03-13 21:07:25.131  INFO 25842 --- [           main] o.a.maven.surefire.booter.ForkedBooter   : Starting ForkedBooter v2.22.1 on jenkins-10.af.pivotal.io with PID 25842 (started by jenkins in /opt/jenkins/data/workspace/spring-cloud-function-master-ci/spring-cloud-function-context)
2019-03-13 21:07:25.134  INFO 25842 --- [           main] o.a.maven.surefire.booter.ForkedBooter   : No active profile set, falling back to default profiles: default
2019-03-13 21:07:25.191 ERROR 25842 --- [           main] o.s.b.d.LoggingFailureAnalysisReporter   : 

***************************
APPLICATION FAILED TO START
***************************

Description:

A component required a bean named 'org.springframework.boot.context.internalConfigurationPropertiesBinder' that could not be found.


Action:

Consider defining a bean named 'org.springframework.boot.context.internalConfigurationPropertiesBinder' in your configuration.
```

**Comment by olegz** [comment]
Interesting, given that it was building just fine few days ago and after upgrading to boot 2.2, so something must have changed in boot snapshot. Will look. . .

**Comment by olegz** [comment]
Just confirmed, it is definitely related to boot 2.2, something happened in last week, so will need to take a look at some point. That said I'll revert back to boot 2.1 regardless since we are planning to release s-c-function 2.1 shortly (few days).

**Comment by olegz** [comment]
I've reverted these two commits fc6c247938fccd11336fd8a210eb656b88491319 7689f8ed609152c778d9da09ac3ac7c005a05c31 and build is back to normal. Will keep the issue open as we obviously need to fix it for boot 2.2 migration down the road

**Comment by olegz** [comment]
The issue was fixed with https://github.com/spring-cloud/spring-cloud-function/commit/bb2f78cb3650b20818f97fe9d06bc9b3cfb4aa7e commit, but since we not really migrating to boot 2.2 yet we started a [2.2.x](https://github.com/spring-cloud/spring-cloud-function/tree/2.2.x) branch as well and will me diligently merging to until we ready to switch master to boot 2.2.
